Add SBOM export support (#16523)
astral-sh/uv: docs/guides/export.md

---
title: Exporting a lockfile
description: Exporting a lockfile to different formats
---

# Exporting a lockfile

uv can export a lockfile to different formats for integration with other tools and workflows. The
`uv export` command supports multiple output formats, each suited to different use cases.

For more details on lockfiles and how they're created, see the
[project layout](../concepts/projects/layout.md) and
[locking and syncing](../concepts/projects/sync.md) documentation.

## Overview of export formats

uv supports three export formats:

- `requirements.txt`: The traditional pip-compatible
  [requirements file format](https://pip.pypa.io/en/stable/reference/requirements-file-format/).
- `pylock.toml`: The standardized Python lockfile format defined in
  [PEP 751](https://peps.python.org/pep-0751/).

> - `CycloneDX`: An industry-standard [Software Bill of Materials (SBOM)](https://cyclonedx.org/)

Reviewer: ```suggestion
- CycloneDX: A [Software Bill of Materials (SBOM)](https://cyclonedx.org/) format
```

  format.

The format can be specified with the `--format` flag:

```console
$ uv export --format requirements.txt
$ uv export --format pylock.toml
$ uv export --format cyclonedx1.5
```

!!! tip

    By default, `uv export` prints to stdout. Use `--output-file` to write to a file for any format:

    ```console
    $ uv export --format requirements.txt --output-file requirements.txt
    $ uv export --format pylock.toml --output-file pylock.toml
    $ uv export --format cyclonedx1.5 --output-file sbom.json
    ```

## `requirements.txt` format

The `requirements.txt` format is the most widely supported format for Python dependencies. It can be
used with `pip` and other Python package managers.

### Basic usage

```console
$ uv export --format requirements.txt
```

The generated `requirements.txt` file can then be installed via `uv pip install`, or with other
tools like `pip`.

!!! note

    In general, we recommend against using both a `uv.lock` and a `requirements.txt` file. The
    `uv.lock` format is more powerful and includes features that cannot be expressed in

>     `requirements.txt`. If you find yourself exporting a `uv.lock` file, consider opening an issue

Author: This is just copied across from [here](https://github.com/astral-sh/uv/blob/512c0ca5edc34f4eb932d5dd59cc86f842a8817c/docs/concepts/projects/sync.md?plain=1#L193-L194), originally added [here](https://github.com/astral-sh/uv/pull/6778/files#diff-bdfe85a7e0c476def2c69fd9e8de130db9d68a453c7412f97e0809b1a8695697) - happy to remove it though if you think it's no longer needed!

    to discuss your use case.

## `pylock.toml` format

[PEP 751](https://peps.python.org/pep-0751/) defines a TOML-based lockfile format for Python
dependencies. uv can export your project's dependency lockfile to this format.

### Basic usage

```console
$ uv export --format pylock.toml
```

## CycloneDX SBOM format

uv can export your project's dependency lockfile as a Software Bill of Materials (SBOM) in CycloneDX
format. SBOMs provide a comprehensive inventory of all software components in your application,
which is useful for security auditing, compliance, and supply chain transparency.

!!! important

>     Support for exporting to CycloneDX is in [preview](../concepts/preview.md), so may be subject to change.

Reviewer: ```suggestion
    Support for exporting to CycloneDX is in [preview](../concepts/preview.md), and may change in any future release.
```
